### PR TITLE
Fix/aut 11/hottext formating

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/AUT-11

There are 4 issues related to formatting options in Hottext interaction:

1. When user added Underline formatting option to text and then creates a hottext choice - underline option clears for it.
2. Also when user uses superscript or subscript formatting option - in Authoring it clears, but correctly shown in Preview.
3. If user wants to select hottext for several words where one of them has any formatting options set, he is unable to do that.
4. If user applies any formatting option for already created hot text answer choice - it is not saved.

Steps to reproduce:

1. Login as admin / item author
2. Go to Items
3. Create a new item and go to Authoring
4. Add a hottext interaction
5. Enter some text and apply Underline formatting option
6. Select the text and click on appeared magic wand

Actual result (1st issue): formatting option for hottext clears
Expected result: all formatting options should stay for hottext choice

7. Enter some text and apply superscript or subscript formatting option
8. Select the text and click on appeared magic wand

Actual result (2nd issue): in Authoring formatting option clears, but correctly shown in Preview
Expected result: formatting option should be correctly presented both in Authoring and Preview

9. Enter some text and apply any formatting option for one word
10. Select one word without formatting option and one with any of formatting options and click on appeared magic wand

Actual result (3d issue): magic wand disappears, but hottext choice is not created
Expected result: hottext choice is created, all applied formatting options display properly.

11. Create a hot text answer choice and apply any formatting option for it
12. Click out of canvas or save and go to Preview

Actual result (4th issue): formatting option is not saved for already add hot text choice
Expected result: hottext shouldn't be editable.

**How to test:**
1. go to Item Authoring
2. add hottextInteraction
3. format text, try to create hottext with html
4. save and go to item Preview, check that it correctly displayed

### CODE REVIEW

**Review by habemuscode — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/helpers/selectionWrapper.js [hunk 1]
```diff
@@ -125,6 +139,30 @@ define(['jquery'], function($) {
                     }
                 }
                 return false;
+            },
+
+            /**
+             * Wraps the current active selection inside the given element
+             * Warning: will lose any event handlers attached to the wrapped nodes!
```
[the remark this answers is not in the document]
Comment (bziondik) [reply views/js/qtiCreator/widgets/helpers/selectionWrapper.js:146]: This allows wrap selection with part of formatted text(with html inside)
like this case
image

[the remark this answers is not in the document]
Comment (AnastasiaArcadia) [reply views/js/qtiCreator/widgets/helpers/selectionWrapper.js:146]: Could we reflect this in jsdoc then?

**Review by AnastasiaArcadia — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not master

**Pull request merged**